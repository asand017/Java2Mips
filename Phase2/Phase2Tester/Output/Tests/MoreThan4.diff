--- ./Phase2Tester/SelfTestCases/MoreThan4.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Phase2Tester/Output/Tests/MoreThan4.out	2017-05-21 21:54:36.647001739 -0700
@@ -1,13 +1,4 @@
-1
-2
-3
-4
-5
-6
-6
-5
-4
-3
-2
-1
-0
+--- Error (ProgramFault) ---
+expecting argument 1 to be an integer, found undefined value
+  Main() line 13
+  MT4.Start(<Heap.0>+0 6) line 19
