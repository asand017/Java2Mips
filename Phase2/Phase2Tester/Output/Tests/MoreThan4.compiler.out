const vmt_MT4
	:MT4.Start
	:MT4.Change

func Main()
	t.0 = HeapAllocZ(4)
	[t.0] = :vmt_MT4
	if t.0 goto :null1
		Error("null pointer")
	null1:
	t.1 = [t.0]
	t.1 = [t.1+0]
	t.2 = call t.1(t.0 6)
	PrintIntS(t.2)
	ret

func MT4.Start(this p1p2p3p4p5p6)
	t.0 = aux
	PrintIntS(t.1)
	PrintIntS(t.1)
	PrintIntS(t.1)
	PrintIntS(t.1)
	PrintIntS(t.1)
	PrintIntS(t.1)
	t.2 = [this]
	t.2 = [t.2+0]
	t.3 = call t.2(this p6p5p4p3p2p1)
	ret aux
func MT4.Change(this p1p2p3p4p5p6)
	PrintIntS(p1)
	PrintIntS(p1p2)
	PrintIntS(p1p2p3)
	PrintIntS(p1p2p3p4)
	PrintIntS(p1p2p3p4p5)
	PrintIntS(p1p2p3p4p5p6)
	ret 0
